junit-platform-engine/src/main/java/org/junit/platform/engine/support/descriptor/FilePosition.java

```diff
@@ -10,32 +10,35 @@
 
 package org.junit.platform.engine.support.descriptor;
 
-import static org.apiguardian.api.API.Status.STABLE;
-
-import java.io.Serializable;
-import java.util.Objects;
 import java.util.Optional;
 
 import org.apiguardian.api.API;
+import org.apiguardian.api.API.Status;
 import org.junit.platform.commons.logging.Logger;
 import org.junit.platform.commons.logging.LoggerFactory;
-import org.junit.platform.commons.util.Preconditions;
 import org.junit.platform.commons.util.StringUtils;
-import org.junit.platform.commons.util.ToStringBuilder;
 
 /**
  * Position inside a file represented by {@linkplain #getLine line} and
  * {@linkplain #getColumn column} numbers.
  *
  * @since 1.0
  */
-@API(status = STABLE, since = "1.0")
-public class FilePosition implements Serializable {
+@API(status = Status.DEPRECATED, since = "5.7")
+public class FilePosition extends org.junit.platform.engine.discovery.FilePosition {
 
 	private static final long serialVersionUID = 1L;
```

Comment:
Where are file positions de-serialized too?

Reply:
We don't use the serialization ourselves. It's meant for third-party tools that send test events over JVM boundaries. Since we didn't change any fields, I think we should be fine.